kedro/config/omegaconf_config.py

```diff
@@ -421,15 +430,24 @@ def _register_new_resolvers(resolvers: dict[str, Callable]) -> None:
                 _config_logger.debug(msg)
                 OmegaConf.register_new_resolver(name=name, resolver=resolver)
 
-    @staticmethod
-    def _check_duplicates(seen_files_to_keys: dict[Path, set[Any]]) -> None:
+    def _check_duplicates(self, key: str, config_per_file: dict[Path, Any]) -> None:
+        if key == "parameters":
+            seen_file_to_keys = {
+                file: self._get_all_keys(OmegaConf.to_container(config, resolve=False))
+                for file, config in config_per_file.items()
+            }
+        else:
+            seen_file_to_keys = {
+                file: set(config.keys()) for file, config in config_per_file.items()
+            }
+
         duplicates = []
 
-        filepaths = list(seen_files_to_keys.keys())
+        filepaths = list(seen_file_to_keys.keys())
```

Comment:
Is the removal of `s` intentional? I think the plural was correct in the first place.